zeppelin-plugins/launcher/k8s-standard/src/main/java/org/apache/zeppelin/interpreter/launcher/K8sRemoteInterpreterProcess.java

```diff
@@ -0,0 +1,379 @@
+    k8sProperties.put("zeppelin.k8s.namespace", kubectl.getNamespace());
+    k8sProperties.put("zeppelin.k8s.interpreter.pod.name", getPodName());
+    k8sProperties.put("zeppelin.k8s.interpreter.container.name", interpreterGroupName.toLowerCase());
+    k8sProperties.put("zeppelin.k8s.interpreter.container.image", containerImage);
+    k8sProperties.put("zeppelin.k8s.interpreter.group.id", interpreterGroupId);
+    k8sProperties.put("zeppelin.k8s.interpreter.group.name", interpreterGroupName);
+    k8sProperties.put("zeppelin.k8s.interpreter.setting.name", interpreterSettingName);
+    k8sProperties.put("zeppelin.k8s.interpreter.localRepo", "/tmp/local-repo");
+    k8sProperties.put("zeppelin.k8s.interpreter.rpc.portRange", String.format("%d:%d", getPort(), getPort()));
+    k8sProperties.put("zeppelin.k8s.server.rpc.host", zeppelinServiceHost);
+    k8sProperties.put("zeppelin.k8s.server.rpc.portRange", zeppelinServiceRpcPort);
+    if (ownerUID() != null && ownerName() != null) {
+      k8sProperties.put("zeppelin.k8s.server.uid", ownerUID());
+      k8sProperties.put("zeppelin.k8s.server.pod.name", ownerName());
+    }
+
+    // environment variables
+    envs.put("SERVICE_DOMAIN", envs.getOrDefault("SERVICE_DOMAIN", System.getenv("SERVICE_DOMAIN")));
+    envs.put("ZEPPELIN_HOME", envs.getOrDefault("ZEPPELIN_HOME", "/zeppelin"));
+
+    if (isSpark()) {
+      int webUiPort = 4040;
+      k8sProperties.put("zeppelin.k8s.spark.container.image", sparkImage);
+      if (isSparkOnKubernetes(properties)) {
+        envs.put("SPARK_SUBMIT_OPTIONS", envs.getOrDefault("SPARK_SUBMIT_OPTIONS", "") + buildSparkSubmitOptions());
+      }
+      envs.put("SPARK_HOME", envs.getOrDefault("SPARK_HOME", "/spark"));
+
+      // configure interpreter property "zeppelin.spark.uiWebUrl" if not defined, to enable spark ui through reverse proxy
+      String webUrl = (String) properties.get("zeppelin.spark.uiWebUrl");
+      if (webUrl == null || webUrl.trim().isEmpty()) {
+        properties.put("zeppelin.spark.uiWebUrl",
+            String.format("//%d-%s.%s", webUiPort, getPodName(), envs.get("SERVICE_DOMAIN")));
+      }
+    }
+
+    k8sProperties.put("zeppelin.k8s.envs", envs);
+
+    // interpreter properties overrides the values
+    k8sProperties.putAll(Maps.fromProperties(properties));
+    return k8sProperties;
+  }
+
+  @VisibleForTesting
+  boolean isSpark() {
+    return "spark".equalsIgnoreCase(interpreterGroupName);
+  }
+
+  boolean isSparkOnKubernetes(Properties interpreteProperties) {
+    String propertySparkMaster = (String) interpreteProperties.getOrDefault("master", "");
+    if (propertySparkMaster.startsWith("k8s://")) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  @VisibleForTesting
+  String buildSparkSubmitOptions() {
+    StringBuilder options = new StringBuilder();
+
+    options.append(" --master k8s://https://kubernetes.default.svc");
+    options.append(" --deploy-mode client");
```

Comment:
With this Cluster mode for spark may not work. Though it may not be needed since the spark-interpreter itself is launched in a new pod. Usually driver memory might require changes as per application being run.

Reply:
Updated to add `--driver-memory` configuration when `spark.driver.memory` is defined.